backtester/portfolio/simpleportfolio.py

```diff
@@ -1,13 +1,34 @@
+import logging
 import math
 from .portfolio import Portfolio
 
+logging.basicConfig(level=logging.DEBUG)
+
 
 class SimplePortfolio(Portfolio):
-    """Allocates all capital to the first signal processed"""
+    """Allocates all capital to the first signal processed.
+    Takes only long positions.
+    Only sells if it has the given symbol in portfolio.
+    """
 
     def __init__(self, *args):
         super().__init__(*args)
 
-    def _get_allocation(self, strength, price):
+    def _get_allocation(self, signal, price):
         """Allocates all capital to the given signal"""
-        return math.floor(self.current_position["Cash"] / price)
+        win_pct, sma = signal.strength
+        if signal.direction == "SELL":
```

Comment:
you could handle these as constants: `event.SELL`, `event.BUY`, `event.MARKET_TYPE`, etc.